frontend/app-development/layout/PageLayout.tsx

```diff
@@ -63,7 +51,27 @@ export const PageLayout = (): React.ReactNode => {
           isRepoError={repoStatusError !== null}
         />
       </PageHeaderContextProvider>
-      {renderPages()}
+      <PagesToRender repoStatus={repoStatus} repoStatusError={repoStatusError} />
     </>
   );
 };
+
+type PagesToRenderProps = {
+  repoStatusError: AxiosError;
+  repoStatus: RepoStatus;
+};
+const PagesToRender = ({ repoStatusError, repoStatus }: PagesToRenderProps) => {
+  // Listen to URL-search params and opens settings-modal if params matches.
+  useOpenSettingsModalBasedQueryParam();
```

Comment:
Nit: suggest adding some space in the component for readability 😅 
```suggestion
  useOpenSettingsModalBasedQueryParam();
  
```